### PR TITLE
docs: adds troubleshoot guidance for delta issues

### PR DESCRIPTION
**Pull request by MuchoLucho**
Includes several places where you can take a look if there is any issue with the delta updates feature

Changelog: None
Ticket: None

Signed-off-by: Luis Ramirez Vargas <luis.ramirez@northern.tech>

# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [ ] Make sure that all commits follow the conventional commit [specification](https://www.github.com/mendersoftware/mendertesting/commitlint/grammar.md) for the Mender project.

The majority of our contributions are fixes, which means your commit should have
the form below:

```
fix: <SHORT DESCRIPTION OF FIX>

<OPTIONAL LONGER DESCRIPTION>

Changelog: <USER-FRIENDLY-CHANGE-DESCRIPTION> or <None>
Ticket: <TICKET NUMBER> or <None>
```

- [ ] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

Please describe your pull request.

Thank you!

### CODE REVIEW

**Comment by TheMeaningfulEngineer** [comment]
@MuchoLucho  Seems like you made a branch directly in the repo instead of your fork. Not much difference practically but was sure this wasn't allowed.

@lluiscampos Don't have much opinion on this, do comment if this needs to come from a fork instead.

**Review by TheMeaningfulEngineer — COMMENTED** [review]
LGTM

Optional - make the example more code based on a concrete example instead of trying to be general.
However even in this shape much better then the previous version.

**Review by TheMeaningfulEngineer — APPROVED** [review]

File: 301.Troubleshoot/03.Mender-Client/docs.md [hunk 1]
```diff
@@ -328,3 +328,33 @@ The block below shows 3 example artifacts.
...
+
+### Checking inconsistencies about the root file system checksums
+
+The checksum shown with `mender show-provides` does not necessarily represent the running partition's actual checksum.
+
+For example, if you remount the partition `rw`, change something and remount it `ro`, the checksum in the `mender provides` won't change, while the checksum of the running partition changes.
+
+As part of your troubleshooting, you can check that the actual checksum on the device is the same as in the `Depends` field.
+
+A general command should be something like the below, but it depends on your device's type of dd.
```
Comment (TheYoctoJester) [inline 301.Troubleshoot/03.Mender-Client/docs.md:340]: "type of dd" sounds like we have seen problems with different implementations of `dd`. Is this accurate? Or did you mean to refer the input file?

File: 301.Troubleshoot/03.Mender-Client/docs.md [hunk 2]
```diff
@@ -328,3 +328,33 @@ The block below shows 3 example artifacts.
...
+
+You can check what is your active root file system (`$ACTIVE_PARTITION` in the following `dd`) with a command like `lsblk`, `fdisk -l`, or similar.
```
Comment (TheYoctoJester) [inline 301.Troubleshoot/03.Mender-Client/docs.md:342]:
```suggestion
You can identify your active root file system (`$ACTIVE_PARTITION` in the following `dd`) with a command like `lsblk`, `fdisk -l`, `mount` or similar.
```

File: 301.Troubleshoot/03.Mender-Client/docs.md [hunk 3]
```diff
@@ -328,3 +328,33 @@ The block below shows 3 example artifacts.
...
+
+The payload size of the version running on the device can be fetched from `mender-artifact read delta.mender`, check for the variable `rootfs_file_size` and use it as `$PAYLOAD_SIZE` in the following `dd`.
```
Comment (TheYoctoJester) [inline 301.Troubleshoot/03.Mender-Client/docs.md:344]:
```suggestion
The payload size of the version running on the device can be obtained from `mender-artifact read delta.mender`, check for the variable `rootfs_file_size` and use it as `$PAYLOAD_SIZE` in the following `dd`.
```

**Review by TheYoctoJester — CHANGES_REQUESTED** [review]
Especially the first comment needs to be sorted out. When reading it, I am uncertain if the `dd` implementation I have is ok (`busybox`, `coreutils`,...)

File: 301.Troubleshoot/03.Mender-Client/docs.md (hunk 1 above)
Comment (TheMeaningfulEngineer) [reply 301.Troubleshoot/03.Mender-Client/docs.md:340]: This originates from me, it was a cheap way to say "It wasn't working with my busybox `dd` but don't know what I did!".
Don't know what the solution was of if there was a workaround or I just installed the correct one.
Voting that we just remove that part and assume real dd.

File: 301.Troubleshoot/03.Mender-Client/docs.md [hunk 4]
```diff
@@ -328,3 +328,33 @@ The block below shows 3 example artifacts.
 * artifact - `mender-artifact read <mender-artifact.mender>`
 * device - Run the command on the device `mender show-provides`
 * server UI - `Releases -> Select Release -> Expand the artifact info by clicking it -> Expand Provides and Depends`
+
+### Checking inconsistencies about the root file system checksums
+
+The checksum shown with `mender show-provides` does not necessarily represent the running partition's actual checksum.
+
+For example, if you remount the partition `rw`, change something and remount it `ro`, the checksum in the `mender provides` won't change, while the checksum of the running partition changes.
```
Comment (lluiscampos) [inline 301.Troubleshoot/03.Mender-Client/docs.md:336]: For consistency,

```suggestion
For example, if you remount the partition `rw`, change something and remount it `ro`, the checksum in the `mender show-provides` won't change, while the checksum of the running partition changes.
```

**Comment by TheYoctoJester** [comment]
Merged the new information as well as old comments in. Thoughts @TheMeaningfulEngineer @MuchoLucho ?

**Review by TheMeaningfulEngineer — APPROVED** [review]
Rendered and went through the structure.
Looks good

**Pull request merged**